### PR TITLE
Updating HLT DQM and validation for Hgg paths

### PR DESCRIPTION
**Pull request by mplaner**
Based a simple HLT DQM on single photon PR here: https://github.com/cms-sw/cmssw/pull/18971

Will need to merge this after the final version of PR 18971 is accepted.

Simple eta/pt plots are enough to monitor performance.

### CODE REVIEW

File: DQMOffline/Trigger/python/HiggsMonitoring_Client_cff.py [hunk 1]
```diff
@@ -1,4 +1,18 @@
 import FWCore.ParameterSet.Config as cms
 
-higgsClient = cms.Sequence(
+diphotonEfficiency = cms.EDAnalyzer("DQMGenericClient",
+    subDirs        = cms.untracked.vstring("HLT/Higgs/*"),
+    verbose        = cms.untracked.uint32(0), # Set to 2 for all messages
+    resolution     = cms.vstring(),
+    efficiency     = cms.vstring(
+        "eff_diphoton         'Photon turnON;             Photon pt [GeV]; efficiency'     photon_pt_numerator          photon_pt_denominator",
+        "eff_diphoton_variable 'Photon turnON;            Photon pt [GeV]; efficiency'     photon_pt_variable_numerator photon_pt_variable_denominator",
```
Comment (mtosi) [inline DQMOffline/Trigger/python/HiggsMonitoring_Client_cff.py]: where is the eff vs eta ?
I would suggest you to add the eff vs phi and eff vs eta-phi
it is important for the HEP 17

Comment (mtosi) [inline DQMOffline/Trigger/python/HiggsMonitoring_Client_cff.py]: and vs di-object mass as well

**Comment by mtosi** [comment]
I think it would be worth to push only code which differs from bhawna's PR

File: DQMOffline/Trigger/python/PhotonMonitor_cff.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import FWCore.ParameterSet.Config as cms
+
+from DQMOffline.Trigger.PhotonMonitor_cfi import hltPhotonmonitoring
+
+# HLT_SinglePhoton200_IDTight
+SinglePhoton200_monitoring = hltPhotonmonitoring.clone()
+SinglePhoton200_monitoring.FolderName = cms.string('HLT/Photon/Photon200/')
+SinglePhoton200_monitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_Photon200_v*")
```
Comment (mtosi) [inline DQMOffline/Trigger/python/PhotonMonitor_cff.py]: are you sure it is available ?

**Comment by mtosi** [comment]
in addition, I'm a little bit puzzle but this
so, you are simply taking single photon trigger monitoring as developed by bhawna
and then you are adding a client (which should be already available)
and update a config file which is run in the VALIDATION step

right ?

File: DQMOffline/Trigger/python/HiggsMonitoring_Client_cff.py (hunk 1 above)
Comment (mplaner) [inline DQMOffline/Trigger/python/HiggsMonitoring_Client_cff.py]: Sorry, I will add the eta plot.  You want also the 2-d eta vs phi efficiency?

The mass plot requires a few changes to the .cc file (discussed in a later comment).

**Comment by mplaner** [comment]
Hi Mia, 
I wasn't sure exactly what the right protocol was since Bwana's PR wasn't accepted yet.  But my plan was to update the PR once those changes have been accepted (and to remove the files which are updated in that PR).  

As for the mass, I am trying to avoid duplicating code; I think the simplest procedure is to add an if nphotons>2 (selection requirement from python), then add code to PhotonMonitor.cc/.h that creates the mass and possibly produce separate plots for lead/sublead efficiency.  

Alternatively, we can create a separate .cc/.h for diphotons and copy what Bwana already has, adding in the mass calculation.

**Comment by mtosi** [comment]
I do not have strong opinion
and I totally agree, please try to exploit and share the same code, if
possible (and it makes sense)
I agree a simple addition of a flag could do the trick

thanks
  mia

**Comment by mtosi** [comment]
I think you need to rebase your branch

**Comment by mplaner** [comment]
Hi Mia,
I did a rebase wrt cmssw-master.  I thought I commented on here for my last update, but I added in the eta, sublead (eta and pt), and mass plots.  The binning for the mass window changes for low mass so that we get good segmentation in the turn on region.

File: DQMOffline/Trigger/plugins/PhotonMonitor.cc [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+  if ( int(photonHandle->size()) < nphotons_ ) return;
+  for ( auto const & p : *photonHandle ) {
+    if ( photonSelection_( p ) ) photons.push_back(p);
+  }
+  if ( int(photons.size()) < nphotons_ ) return;
+  
+  // filling histograms (denominator)  
+
+  photonME_.denominator -> Fill(photons[0].pt());
+<<<<<<< HEAD
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/PhotonMonitor.cc]: uhmmm, I think you still need to clean the code :(

File: DQM/Integration/python/clients/beam_dqm_sourceclient-live_cfg.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import FWCore.ParameterSet.Config as cms
```
Comment (mtosi) [inline DQM/Integration/python/clients/beam_dqm_sourceclient-live_cfg.py]: are you sure this changes belong to your developments for trigger DQM/validation ?

File: DQM/Integration/python/clients/beampixel_dqm_sourceclient-live_cfg.py [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import FWCore.ParameterSet.Config as cms
```
Comment (mtosi) [inline DQM/Integration/python/clients/beampixel_dqm_sourceclient-live_cfg.py]: are you sure this changes belong to your developments for trigger DQM/validation ?

File: DQM/Integration/python/config/fileinputsource_cfi.py [hunk 6]
```diff
@@ -78,6 +78,7 @@
 runstr = str(options.runNumber)
```
Comment (mtosi) [inline DQM/Integration/python/config/fileinputsource_cfi.py]: are you sure this changes belong to your developments for trigger DQM/validation ?

File: RecoPixelVertexing/PixelTrackFitting/python/PixelTracks_2017_cff.py [hunk 7]
```diff
@@ -0,0 +1,134 @@
+import FWCore.ParameterSet.Config as cms
```
Comment (mtosi) [inline RecoPixelVertexing/PixelTrackFitting/python/PixelTracks_2017_cff.py]: are you sure this changes belong to your developments for trigger DQM/validation ?

**Comment by mtosi** [comment]
did you do 
`git rebase -i CMSSW_9_2_X` ?

**Comment by slava77** [comment]
-1
needs a rebase (I'm guessing that 920patch2 or later was used instead of a 92X IB to make this PR)

**Comment by mplaner** [comment]
I have rebased and recompiled all changes starting from the may 30th nightly build.  I submitted a new PR to supersede this one here:

https://github.com/cms-sw/cmssw/pull/19023

**Pull request closed without merging**